docs/concepts/style_guide.md

```diff
@@ -0,0 +1,178 @@
+# Beaker Style Guide
+
+## Scope of this guide
+
+The purpose of this guide is to provide definitions for preferred practices when
+writing Beaker tests, both syntactically and stylistically. This guide will
+define and provide examples for preferred test layout and conventions. Common
+patterns that are recommend as well as patterns that should be avoided will be
+described.
+
+No style manual can cover every possible circumstance. When a judgement call
+becomes necessary, keep in mind the following general ideas:
+
+1. **Readability matters**. If you have to choose between two equally effective
+alternatives, pick the more readable one. This is, of course, subjective, but if
+you can read your own code three months from now, that’s a great start. Don’t be
+clever over readable, unless you have a documented purpose. Use object oriented
+programming when things get complex and DRY.
+2. **Inherit upstream conventions**. Beaker is still ruby, so use the
+[ruby community style guide](https://github.com/bbatsov/ruby-style-guide). When
+not called out here, use the ruby style guide.
+
+## Test Naming
+
+Tests should test what they say they test. Test names, both the name of the test
+file and the value given to the
+[`test_name`](http://www.rubydoc.info/github/puppetlabs/beaker/Beaker/DSL/Structure#test_name-instance_method)
+function should provide an accurate indication about the purpose of the test.
+
+The `test_name` function should be the first line in the Beaker test file.
+
+**Good:**
+```ruby
+# head -1 puppet/acceptance/tests/resource/file/should_default_mode.rb
+test_name "file resource: set default modes"
+```
+
+## Structure Methods Should Use Explicit Blocks
+
+These methods aid in self-documenting your tests, including indention in the
+logs. If you don’t use explicit blocks, beaker does not know how to properly
+indent your test’s output.
+
+The most common
+[structure methods](http://www.rubydoc.info/github/puppetlabs/beaker/Beaker/DSL/Structure)
+are
+[`#test_name`](http://www.rubydoc.info/github/puppetlabs/beaker/Beaker/DSL/Structure#test_name-instance_method),
+[`#step`](http://www.rubydoc.info/github/puppetlabs/beaker/Beaker/DSL/Structure#step-instance_method),
+and
+[`#teardown`](http://www.rubydoc.info/github/puppetlabs/beaker/Beaker/DSL/Structure#teardown-instance_method).
+
+**Good:**
+```ruby
+step 'do this thang' do
+  on(host, "echo 'do this thang'")
+end
+```
+**Bad:**
+```ruby
+step 'do this thang'
+on(host, "echo 'do this thang'")
+```
+
```

Comment:
In the future, we should include opinionated examples of dealing with variable scoping in the context of using blocked structure methods.